drivers/mtd_spi_nor/mtd_spi_nor.c

```diff
@@ -622,19 +659,50 @@ static int mtd_spi_nor_write_page(mtd_dev_t *mtd, const void *src, uint32_t page
     /* write enable */
     mtd_spi_cmd(dev, dev->params->opcode->wren);
 
+    /* Wait for WEL-Flag to be set */
+    wait_for_write_enable_set(dev);
+
     /* Page program */
     mtd_spi_cmd_addr_write(dev, dev->params->opcode->page_program, addr, src, size);
 
     /* waiting for the command to complete before returning */
     wait_for_write_complete(dev, 0);
 
+#if IS_USED(MODULE_MTD_SPI_NOR_MX_SECURITY)
+    /* Read security register */
+    uint8_t scur_reg;
+    mtd_spi_cmd_read(dev, dev->params->opcode->rdscur, &scur_reg, sizeof(scur_reg));
+    if (scur_reg & MX_SECURITY_PFAIL) {
+        DEBUG("mtd_spi_nor_write: ERR: page program failed! scur = %02x\n", scur_reg);
+        ret = -EIO;
+    }
+#elif IS_USED(MODULE_MTD_SPI_NOR_ISSI_SECURITY)
```

Comment:
Since we know the vendor from the device ID, can we do this check at run-time?

Reply:
That is absolutely possible, I decided against it to avoid having too much "dead" code. Considering how different Macronix and ISSI handle that stuff, ST, Microchip, Winbond, ... probably handle it different as well, so we would have many checks left in the final binary which are never needed in normal operation.

Considering so many other parameters of the Flash have to be known at compile time, it didn't make sense to me to check it on runtime.

HOWEVER now that I think about it, it is probably possible to add a .manufacturer variable to the mtd_spi_nor_params_t struct. That would get rid of the pseudomodules and the compiler can optimize the code at compile time, because the variable is already set.

Reply:
> Considering so many other parameters of the Flash have to be known at compile time, it didn't make sense to me to check it on runtime.

Actually there aren't any *critical* ones - the timings that get set are just minimal values, the size is determined at run time. The idea is that you can just attach any flash chip as those often get swapped out between board revisions depending on availability.